index.bs

```diff
@@ -3076,6 +3091,57 @@ and |session|:
 
 </div>
 
+<div algorithm="locate nodes using accessibility attributes">
+To <dfn>locate nodes using accessibility attributes</dfn> with given |context nodes|, |selector|,
+|maximum returned node count|, and |session|:
+
+1. If |selector|'s <code>role</code> and |selector|'s <code>name</code> are missing, return [=error=] with [=error code=]
+   [=invalid selector=].
+
+1. Let |returned nodes| be an empty [=/list=].
+
+1. For each |context node| in |context nodes|:
+
+  1. Let |match| be true.
+
+  1. If |selector|'s <code>role</code> is present:
+
+    1. Let |role| be the result of computing the [=accessible role=] of |context node|.
+
+    1. If |selector|'s <code>role</code> [=is|is not=] |role|:
+
+      1. Set |match| to false.
+
+  1. If |selector|'s <code>name</code> is present:
+
+    1. Let |name| be the result of computing the [=accessible name=] of |context node|.
+
+    1. If |selector|'s <code>name</code> [=is|is not=] |name|:
+
+      1. Set |match| to false.
+
+  1. If |match| is true:
+
+    1. [=list/Append=] |context node| to |returned nodes|.
+
+  1. Let |child nodes| be an empty [=/list=] and, for each node |child| in the
+     <a spec=dom>children</a> of |context node|:
+
+    1. If |child| implements {{Element}}, [=list/append=] |child| to |child nodes|.
+
+  1. Let |child node matches| be the result of [=locate nodes using accessibility attributes=]
+     with |child nodes|, |selector|, |maximum returned node count|, and |session|.
+
+  1. [=list/Extend=] |returned nodes| with |child node matches|.
+
+1. If |maximum returned node count| is not null, [=list/remove=] all entries
+   in |returned nodes| with an index greater than or equal to |maximum returned
+   node count|.
+
```

Comment:
When seeing these lines again I actually wonder if that behavior is correct. Consider a node with 2 children whereby the first children has a lot of children as well. If we have collected more matching nodes as allowed, the 2nd child of the original context node wouldn't be included.

I wonder if we should flatten the array at the very end from top to bottom to not actually loose important already collected nodes. It's not blocking this PR from landing but we should at least file a new issue to further discuss (if necessary) and fix.

Reply:
I think you are right and it looks like the innerText selector has the same issue, let me look over it again

Reply:
I reopened this issue for awaiting the results of the check.

Reply:
I have rewrote that part also addressing this issue (marked wrong thread).